File: week12/week12-3.cpp
///week12-3.cpp
#include <stdio.h>
int isPrime(int n)
{
    for(int i=2;i<n;i++) {
        if(n%i==0) return 0;
    }
    return 1;
}
int main()
{
    printf("請輸入一個數:");
    int n;
    scanf("%d",&n);
    int ans=0;
    for(int i=2;i<=n;i++) {
        if(isPrime(i)){
            printf("%d ",i);
            ans ++;
        }
    }
    printf("總共有 %d 個質數\n",ans);
}

File: week12/week12-3.py
#week12-3.py 質數篩子法
table= [True]*240000
ans=0 #有幾個質數
for i in range(2,240000): #找出所有可能的質數，質數不會被整除，寶格裡還留著繼承權
  if table[i]==True: #找到質數
    ans += 1
    for k in range(i*i,240000,i): table[k]=False
print("找到",ans,'個質數')